p2p/src/net/mock/transport/impls/stream_adapter/noise.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+use futures::future::BoxFuture;
+use snowstorm::NoiseStream;
+use tokio::time::timeout;
+
+use crate::{
+    error::P2pError,
+    net::mock::{peer::Role, transport::PeerStream},
+};
+
+use super::StreamAdapter;
+
+// How much time is allowed to spend setting up (optionally) encrypted stream.
+const HANDSHAKE_TIMEOUT: Duration = Duration::from_secs(10);
+
+static NOISE_HANDSHAKE_PATTERN: &str = "Noise_XX_25519_ChaChaPoly_SHA256";
+
+static NOISE_HANDSHAKE_PARAMS: once_cell::sync::Lazy<snowstorm::NoiseParams> =
+    once_cell::sync::Lazy::new(|| NOISE_HANDSHAKE_PATTERN.parse().expect("valid pattern"));
```

Comment:
Noise protocol names are well known and documented in the spec: https://noiseprotocol.org/noise.html#protocol-names-and-modifiers
So I think it's OK to parse it. lib-p2p also parses it from the string.

Reply:
My motivation for pointing this out was that you could avoid `once_cell` and such and make it just a `const`. Your call in the end, I don't think it's a big deal

Reply:
Right, static initialization would be better here.
But `NoiseParams` has some annoying information duplication. It requires both text noise protocol name and protocol details. So we will need to make sure that the protocol name and actual values are in sync:

```
pub struct NoiseParams {
    pub name:      String,
    pub base:      BaseChoice,
    pub handshake: HandshakeChoice,
    pub dh:        DHChoice,
    #[cfg(feature = "hfs")]
    pub kem:       Option<KemChoice>,
    pub cipher:    CipherChoice,
    pub hash:      HashChoice,
}
```

Also we won't be able const struct because of `String` .

Reply:
Ok, that's quite annoying. Let's leave it as is with the string parsing then.